source/material_model/visco_plastic.cc

```diff
@@ -347,13 +375,55 @@ namespace aspect
             out.reaction_terms[i][c] = 0.0;
           // If strain weakening is used, overwrite the first reaction term,
           // which represents the second invariant of the strain tensor
-          if  (use_strain_weakening && this->get_timestep_number() > 0)
+          double edot_ii = 0.;
+          double e_ii = 0.;
+          if  (use_strain_weakening == true && use_finite_strain_tensor == false && this->get_timestep_number() > 0)
             {
-              const double edot_ii = std::max(sqrt(std::fabs(second_invariant(deviator(in.strain_rate[i])))),min_strain_rate);
-
+              edot_ii = std::max(sqrt(std::fabs(second_invariant(deviator(in.strain_rate[i])))),min_strain_rate);
+              e_ii = edot_ii*this->get_timestep();
               // Update reaction term
-              out.reaction_terms[i][0] = edot_ii*this->get_timestep();
+              out.reaction_terms[i][0] = e_ii;
+            }
+        }
+
+      // We need the velocity gradient for the finite strain (they are not included in material model inputs),
+      // so we get them from the finite element.
+      if (in.cell && use_strain_weakening == true && use_finite_strain_tensor == true && this->get_timestep_number() > 0)
+        {
+          const QGauss<dim> quadrature_formula (this->get_fe().base_element(this->introspection().base_elements.velocities).degree+1);
+          FEValues<dim> fe_values (this->get_mapping(),
+                                   this->get_fe(),
+                                   quadrature_formula,
+                                   update_gradients);
+
+          std::vector<Tensor<2,dim> > velocity_gradients (quadrature_formula.size(), Tensor<2,dim>());
+
+          fe_values.reinit (*in.cell);
+          fe_values[this->introspection().extractors.velocities].get_function_gradients (this->get_solution(),
+                                                                                         velocity_gradients);
```

Comment:
Sorry, should have quoted lines 394-397, but the build error there propagates down to these lines as well.